### PR TITLE
az configure: supports configuring local defaults 

### PR DESCRIPTION
**Pull request by yugangw-msft**
CC @limingu  @panchagnula 

To do it in command:
```
    az configure --defaults web=web1 --scope local
    az configure --list-defaults --scope local
```

To do it in code
```python
   with ConfiguredDefaultSetter(cmd.cli_ctx.config, True):
       cmd.cli_ctx.config.set_value('defaults', 'group', rg_name)
       cmd.cli_ctx.config.set_value('defaults', 'sku', full_sku)
       cmd.cli_ctx.config.set_value('defaults', 'plan', asp)
       cmd.cli_ctx.config.set_value('defaults', 'location', location)
       cmd.cli_ctx.config.set_value('defaults', 'web', name)
```

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
Thanks @yugangw-msft this looks good to me. Once this is in dev I will update up to use this as well :)

File: src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py [hunk 1]
```diff
@@ -107,54 +109,49 @@ def _handle_global_configuration(config):
...
-                logger.error('TTL must be a positive integer')
-                cache_ttl = None
-        # save the global config
-        config.set_value('core', 'output', OUTPUT_LIST[output_index]['name'])
-        config.set_value('core', 'collect_telemetry', 'yes' if allow_telemetry else 'no')
-        config.set_value('core', 'cache_ttl', cache_ttl)
-        config.set_value('logging', 'enable_log_file', 'yes' if enable_file_logging else 'no')
-        if need_to_reset_use_local_config:
-            config.set_to_use_local_config(True)
+        with ConfiguredDefaultSetter(config):
```
Comment (limingu) [inline src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py]: Here is to set global only. So we may want to set as.

with ConfiguredDefaultSetter(config, False):

File: src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py [hunk 2]
```diff
@@ -107,54 +109,49 @@ def _handle_global_configuration(config):
...
+                cmd.cli_ctx.config.set_value(defaults_section, parts[0], _normalize_config_value(parts[1]))
         return
     if list_defaults:
-        use_local_config_original = cmd.cli_ctx.config.use_local_config
-        cmd.cli_ctx.config.set_to_use_local_config(list_defaults == 'local')
-        defaults_result = cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)
-        cmd.cli_ctx.config.set_to_use_local_config(use_local_config_original)
-        return defaults_result
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, scope.lower() == 'local'):
+            defaults_result = cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)
```
Comment (limingu) [inline src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py]: Why not just:
return cmd.cli_ctx.config.items(cmd.cli_ctx.config.defaults_section_name)

File: src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/tests/latest/test_configure.py [hunk 3]
```diff
@@ -21,5 +25,37 @@ def test_configure_output_options(self):
...
+
+    def setUp(self):
+        self.local_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        self.cmd('configure --defaults global="" global2=""')
+        self.cmd('configure --defaults local="" --scope local')
+
+    def test_configure_global_defaults(self):
+        import os
```
Comment (limingu) [inline src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/tests/latest/test_configure.py]: There is already import os. This line can be removed.

File: src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py (hunk 2 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py]: we would like only display  valid defaults which have to have values

Comment (limingu) [reply src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py]: Ok, good to know we have some keys that didn't have any values.

**Pull request merged**